### PR TITLE
[ur] Add Queue testsuite

### PR DESCRIPTION
**Pull request by veselypeta**
Implements queue conformance testsuite.

### CODE REVIEW

File: test/conformance/queue/urQueueGetInfo.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+UUR_TEST_SUITE_P(urQueueGetInfoTest,
+                 ::testing::Values(
+
+                     UR_QUEUE_INFO_CONTEXT,         //
+                     UR_QUEUE_INFO_DEVICE,          //
+                     UR_QUEUE_INFO_DEVICE_DEFAULT,  //
+                     UR_QUEUE_INFO_PROPERTIES,      //
+                     UR_QUEUE_INFO_REFERENCE_COUNT, //
+                     UR_QUEUE_INFO_SIZE             //
```
Comment (kbenzie) [inline test/conformance/queue/urQueueGetInfo.cpp:15]: How does `clang-format` deal with this change?

```suggestion
                     UR_QUEUE_INFO_CONTEXT,
                     UR_QUEUE_INFO_DEVICE,
                     UR_QUEUE_INFO_DEVICE_DEFAULT,
                     UR_QUEUE_INFO_PROPERTIES,
                     UR_QUEUE_INFO_REFERENCE_COUNT,
                     UR_QUEUE_INFO_SIZE,
```

I might look less weird.

File: test/conformance/queue/urQueueFinish.cpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    ur_mem_handle_t buffer = nullptr;
+    ASSERT_SUCCESS(urMemBufferCreate(context, UR_MEM_FLAG_READ_WRITE,
+                                     buffer_size, nullptr, &buffer));
+
+    std::vector<uint8_t> data(buffer_size, 42);
+    ASSERT_SUCCESS(urEnqueueMemBufferWrite(queue, buffer, /* blocking */ false,
+                                           0, 1024, data.data(), 0, nullptr,
+                                           nullptr));
+
+    ASSERT_SUCCESS(urQueueFinish(queue));
```
Comment (fabiomestre) [inline test/conformance/queue/urQueueFinish.cpp:19]: Should we validate that the enqueued tasks have actually finished after calling urQueueFinish? The function could return success but actually do nothing.

File: test/conformance/queue/urQueueFinish.cpp [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+TEST_P(urQueueFinishTest, InvalidNullHandleQueue) {
+    ASSERT_EQ_RESULT(UR_RESULT_ERROR_INVALID_NULL_HANDLE,
```
Comment (fabiomestre) [inline test/conformance/queue/urQueueFinish.cpp:23]: Is UR_RESULT_ERROR_INVALID_QUEUE something that could be tested as well?

File: test/conformance/queue/urQueueGetInfo.cpp (hunk 1 above)
Comment (veselypeta) [reply test/conformance/queue/urQueueGetInfo.cpp:15]: It does this
```cpp
UUR_TEST_SUITE_P(urQueueGetInfoTest,
                 ::testing::Values(UR_QUEUE_INFO_CONTEXT, UR_QUEUE_INFO_DEVICE,
                                   UR_QUEUE_INFO_DEVICE_DEFAULT,
                                   UR_QUEUE_INFO_PROPERTIES,
                                   UR_QUEUE_INFO_REFERENCE_COUNT,
                                   UR_QUEUE_INFO_SIZE),
                 uur::deviceTestWithParamPrinter<ur_queue_info_t>);
```
The reason I added the comments was to have one enum per line, but I think this looks fine.

File: test/conformance/queue/urQueueFinish.cpp (hunk 3 above)
Comment (veselypeta) [reply test/conformance/queue/urQueueFinish.cpp:23]: This is just the tests that we are porting from UnitUR. Some of them aren't as complete as we would like. I have #223 to track it.

File: test/conformance/queue/urQueueFinish.cpp (hunk 2 above)
Comment (veselypeta) [reply test/conformance/queue/urQueueFinish.cpp:19]: Finish will block until all commands to complete.

File: test/conformance/queue/urQueueRelease.cpp [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: MIT
+#include <uur/fixtures.h>
+
+using urQueueReleaseTest = uur::urQueueTest;
+UUR_INSTANTIATE_DEVICE_TEST_SUITE_P(urQueueReleaseTest);
+
+TEST_P(urQueueReleaseTest, Success) {
+    ASSERT_SUCCESS(urQueueRetain(queue));
```
[the remark this answers is not in the document]
Comment (veselypeta) [reply test/conformance/queue/urQueueRelease.cpp:9]: Good idea. I'm trying to keep this PR just UnitUR tests. I'll create a task to add this test.

[the remark this answers is not in the document]
Comment (veselypeta) [reply test/conformance/queue/urQueueRelease.cpp:9]: #248

File: test/conformance/queue/urQueueFinish.cpp (hunk 2 above)
Comment (veselypeta) [reply test/conformance/queue/urQueueFinish.cpp:19]: Ah sorry I misunderstood your comment. Yes you're right - I'll add a check. I assume you can just query the status of the event right?

Comment (fabiomestre) [reply test/conformance/queue/urQueueFinish.cpp:19]: yeah, I think that would be a good way to do it :)

Comment (veselypeta) [reply test/conformance/queue/urQueueFinish.cpp:20]: done.

File: test/conformance/queue/urQueueGetInfo.cpp (hunk 1 above)
Comment (veselypeta) [reply test/conformance/queue/urQueueGetInfo.cpp]: done

Comment (kbenzie) [reply test/conformance/queue/urQueueGetInfo.cpp]: I think this misses the point I was making which is to have a trailing `,` after the last item in the list. It changes the behaviour of `clang-format`.

Comment (kbenzie) [reply test/conformance/queue/urQueueGetInfo.cpp]: Although I just tried it and it did not do what I expected. Perhaps some interaction with the macro. Nevermind.

**Review by kbenzie — APPROVED** [review]
LGTM

File: test/conformance/queue/urQueueGetInfo.cpp (hunk 1 above)
Comment (veselypeta) [reply test/conformance/queue/urQueueGetInfo.cpp]: Yeah it doesn't like trailing commas - I had to remove it to get it to compile

**Pull request merged**